### PR TITLE
[RPA 750] Highlight the process that caught an error

### PR DESCRIPTION
**Pull request by MRSIMM0**
## Description

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-common/src/bpmn/tasks.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+  EXCLUSIVE_GATEWAY = 'bpmn:ExclusiveGateway',
+  INCLUSIVE_GATEWAY = 'bpmn:InclusiveGateway',
+  COMPLEX_GATEWAY = 'bpmn:ComplexGateway',
+  ASSOCIATION = 'bpmn:Association',
+  SEQUENCE_FLOW = 'bpmn:SequenceFlow',
+  ITERMEDIATE_THROW_EVENT = 'bpmn:IntermediateThrowEvent',
+  TASK = 'bpmn:Task',
+  BOUNDARY_ERROR = 'bpmn:ErrorEventDefinition'
+
+
```
Comment (preludium) [inline runbotics/runbotics-common/src/bpmn/tasks.ts]: redundant empty space at the end of enum + please fix typo in `ITERMEDIATE_THROW_EVENT` and in the appearances

File: runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts [hunk 2]
```diff
@@ -50,7 +51,13 @@ export class RuntimeSubscriptionsService {
                         event.activity.owner as IActivityOwner
                     ).behaviour;
                     switch (event.activity.content.type) {
-                        case 'bpmn:ServiceTask':
+
+                        case 'bpmn:ErrorEventDefinition': //with BpmnElementType.BOUNDARY_ERROR enum it does not work XD
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts]: wut, nah it has to work. Try to solve it, if there will be no progress then contact me

File: runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts [hunk 3]
```diff
@@ -50,7 +51,13 @@ export class RuntimeSubscriptionsService {
...
+                            processInstanceEvent.log = `ErrorBoundaryEvent: ${event.activity.content.type} ${event.eventType}`;
+                            processInstanceEvent.step = 'event.warning';
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts]: the step should be like `event.errorBoundary`, just to be clear everywhere

File: runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts [hunk 4]
```diff
@@ -62,19 +69,19 @@ export class RuntimeSubscriptionsService {
                                 processInstanceEvent.step = script;
                             }
                             break;
-                        case 'bpmn:EndEvent':
+                        case  BpmnElementType.END_EVENT:
                             processInstanceEvent.step = 'event.end';
                             processInstanceEvent.log = `Activity: ${event.activity.content.type} ${event.eventType}`;
                             break;
-                        case 'bpmn:StartEvent':
+                        case  BpmnElementType.START_EVENT:
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts:75]: you leave a lot of empty spaces/lines in your code. Please pay attention to it

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Label/Label.tsx [hunk 5]
```diff
@@ -1,24 +1,34 @@
...
+
+
 
 export type Color = 'primary' | 'secondary' | 'error' | 'warning' | 'info' | 'success';
 
 interface LabelProps {
     className?: string;
     color?: Color;
     children?: ReactNode;
+    warning?:boolean
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Label/Label.tsx]: space after colon!

File: runbotics/runbotics-orchestrator-ui/src/src-app/components/Label/Label.tsx [hunk 6]
```diff
@@ -1,24 +1,34 @@
...
-        {children}
-    </Wrapper>
+    <LabelGroup>
+        <Wrapper
+            color={color}
+            {...rest}
+        >
+            {children}
+        </Wrapper>
+        {warning && isProcessInstanceFinished(ProcessInstanceStatus[children.toString()])?
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/components/Label/Label.tsx]: `children.toString()` seems like a hack, rewrite it correctly

File: runbotics/runbotics-scheduler/src/websocket/bot/process-launch/bot-process-instance-event.service.ts [hunk 7]
```diff
@@ -26,6 +26,10 @@ const COMPLETED_UPDATE_FIELDS = [
 ];
 const STARTED_UPDATE_FIELDS = ['input', 'created'];
 
+function setWarning(processInstanceEvent: IProcessInstanceEvent): boolean {
+    return processInstanceEvent.step === 'event.warning';
```
Comment (preludium) [inline runbotics/runbotics-scheduler/src/websocket/bot/process-launch/bot-process-instance-event.service.ts]: IMO we should have a single source of truth for elements like `event.warning`

Comment (preludium) [reply runbotics/runbotics-scheduler/src/websocket/bot/process-launch/bot-process-instance-event.service.ts]: the name `setWarning` sounds like a setter in a class or manipulation of the parameter. A better term would be `hasWarning` (in the ideal world I would include the name `processInstance` in it but we can deduct if from parameter)

File: runbotics/runbotics-scheduler/src/websocket/bot/process-launch/bot-process-instance-event.service.ts [hunk 8]
```diff
@@ -177,10 +192,18 @@ export class BotProcessEventService {
             await queryRunner.manager
                 .createQueryBuilder()
                 .update(ProcessInstanceEntity)
-                .set({
-                    error: processInstanceEvent.error,
-                    step: processInstanceEvent.step,
-                })
+                .set(
+                    setWarning(processInstanceEvent)
+                        ? {
```
Comment (preludium) [inline runbotics/runbotics-scheduler/src/websocket/bot/process-launch/bot-process-instance-event.service.ts]: manipulate just a warning field with this function
`
{
    error: processInstanceEvent.error,
    step: processInstanceEvent.step,
    warning: hasWarning(processInstanceEvent.step),
}
`

File: runbotics-orchestrator/.jhipster/ProcessInstance.json [hunk 9]
```diff
@@ -34,7 +34,11 @@
     {
       "fieldName": "step",
       "fieldType": "String"
-    }
+    },
+      {
+          "fieldName": "warning",
+          "fieldType": "Boolean"
+      }
```
Comment (preludium) [inline runbotics-orchestrator/.jhipster/ProcessInstance.json]: redundant, we don't use it. It was relevant only at the project initialization

**Review by preludium — CHANGES_REQUESTED** [review]
It's missing database migration

File: runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts (hunk 2 above)
Comment (MRSIMM0) [reply runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts]: forgot to delete that comment

File: runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230615123802_changelog.xml [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-4.1.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="samelakk" id="20230615123802">
```
Comment (preludium) [inline runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230615123802_changelog.xml]: are you `samelakk` or `pasiecznys`? xd

File: runbotics/runbotics-common/src/model/api/process-instance.model.ts [hunk 11]
```diff
@@ -38,6 +39,12 @@ export enum ProcessInstanceStatus {
     TERMINATED = "TERMINATED",
 }
 
+export enum ProcessInstanceStep {
+    START = 'event.start',
+    END = 'event.end',
+    WARNING = 'event.warning'
```
Comment (preludium) [inline runbotics/runbotics-common/src/model/api/process-instance.model.ts]: not warning but errorBoundary

File: runbotics/runbotics-common/src/bpmn/tasks.ts [hunk 12]
```diff
@@ -1,14 +1,15 @@
...
+  ACTIVITY = 'bpmn:Activity',
+  SERVICE_TASK = 'bpmn:ServiceTask',
+  EXCLUSIVE_GATEWAY = 'bpmn:ExclusiveGateway',
+  INCLUSIVE_GATEWAY = 'bpmn:InclusiveGateway',
+  COMPLEX_GATEWAY = 'bpmn:ComplexGateway',
+  ASSOCIATION = 'bpmn:Association',
+  SEQUENCE_FLOW = 'bpmn:SequenceFlow',
+  INTERMEDIATE_THROW_EVENT = 'bpmn:IntermediateThrowEvent',
+  TASK = 'bpmn:Task',
+  BOUNDARY_ERROR = 'bpmn:errorBoundary'
```
Comment (preludium) [inline runbotics/runbotics-common/src/bpmn/tasks.ts]: it should be ` BOUNDARY_ERROR = 'bpmn:ErrorEventDefinition'`

File: runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts [hunk 13]
```diff
@@ -50,7 +52,11 @@ export class RuntimeSubscriptionsService {
                         event.activity.owner as IActivityOwner
                     ).behaviour;
                     switch (event.activity.content.type) {
-                        case 'bpmn:ServiceTask':
+                        case 'bpmn:ErrorEventDefinition':
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/websocket/bpmn/runtime-subscriptions.service.ts]: use enum

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process.json [hunk 14]
```diff
@@ -639,6 +639,7 @@
     "Process.List.Table.Header.Scheduled": "Zaplanowanego",
     "Process.List.Table.Header.Actions": "Akcje",
     "Process.Details.Modeler.Actions.Event.Start.Label": "Zdarzenie początkowe",
+    "Process.Details.Modeler.Actions.Event.Warning.Label": "Ostrzerzenie",
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process.json]: ostrzeżenie*

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230615123802_changelog.xml (hunk 10 above)
Comment (MRSIMM0) [reply runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230615123802_changelog.xml]: My mistake XD

**Review by preludium — DISMISSED** [review]

**Review by preludium — APPROVED** [review]

**Pull request merged**